File: app/src/main/java/com/example/makemuscle/TimerManager.java
package com.example.makemuscle;

import android.content.Intent;
import android.os.CountDownTimer;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class TimerManager extends AppCompatActivity{
    private CountDownTimer countDownTimer;
    private final TextView mtextView;
    private final Button startBtn;
    private long mTimeLeftInMillis;
    private boolean mTimeRunning;
    private final String buttonValue;

    public TimerManager(TextView textView, Button button, String value) {
        this.mtextView = textView;
        this.startBtn = button;
        this.buttonValue = value;
    }

    public void stopTimer() {
        if (mTimeRunning) {
            if (countDownTimer != null) {
                countDownTimer.cancel();
            }
            mTimeRunning = false;
            startBtn.setText("Start");
        }
    }

    public void startTimer() {
        if (countDownTimer != null) {
            countDownTimer.cancel();
        }

        final CharSequence value1 = mtextView.getText();
        String num1 = value1.toString();
        String num2 = num1.substring(0,2);
        String num3 = num1.substring(3,5);

        final int number = Integer.parseInt(num2) * 60 + Integer.parseInt(num3);
        mTimeLeftInMillis = number * 1000L;

        countDownTimer = new CountDownTimer(mTimeLeftInMillis, 1000) {
            @Override
            public void onTick(long millisUntilFinished) {
                mTimeLeftInMillis = millisUntilFinished;
                updateTimer();
            }

            @Override
            public void onFinish() {
                int newvalue = Integer.parseInt(buttonValue) + 1;
                if (newvalue <= 7) {
                    Intent intent = new Intent(getApplicationContext(), getClass());
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
                    intent.putExtra("value", String.valueOf(newvalue));
                    mtextView.getContext().startActivity(intent);
                } else {
                    newvalue = 1;
                    Intent intent = new Intent(getApplicationContext(), getClass());
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
                    intent.putExtra("value", String.valueOf(newvalue));
                    mtextView.getContext().startActivity(intent);
                }
            }
        }.start();
        startBtn.setText("Pause");
        mTimeRunning = true;
    }

    public boolean isRunning() {
        return mTimeRunning;
    }

    private void updateTimer() {
        int minutes = (int) (mTimeLeftInMillis / 1000) / 60;
        int seconds = (int) (mTimeLeftInMillis / 1000) % 60;

        String timeLeftText = String.format("%02d:%02d", minutes, seconds);
        mtextView.setText(timeLeftText);
    }
}

File: app/src/main/java/com/example/makemuscle/TimerManager.kt
package com.example.makemuscle

import android.content.Context
import android.content.Intent
import android.os.CountDownTimer
import android.widget.Button
import android.widget.TextView

class TimerManager(
    private val mtextView: TextView,
    private val startBtn: Button,
    private val buttonValue: String,
    private val context: Context
) {
    private var countDownTimer: CountDownTimer? = null
    private var mTimeLeftInMillis: Long = 0
    private var mTimeRunning = false

    fun stopTimer() {
        if (mTimeRunning) {
            countDownTimer?.cancel()
            mTimeRunning = false
            startBtn.text = "Start"
        }
    }

    fun startTimer() {
        countDownTimer?.cancel()

        val value1 = mtextView.text
        val num1 = value1.toString()
        val num2 = num1.substring(0, 2)
        val num3 = num1.substring(3, 5)

        val number = Integer.parseInt(num2) * 60 + Integer.parseInt(num3)
        mTimeLeftInMillis = number * 1000L

        countDownTimer = object : CountDownTimer(mTimeLeftInMillis, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                mTimeLeftInMillis = millisUntilFinished
                updateTimer()
            }

            override fun onFinish() {
                var newvalue = buttonValue.toInt() + 1
                if (newvalue <= 7) {
                    val intent = Intent(context, context.javaClass)
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
                    intent.putExtra("value", newvalue.toString())
                    context.startActivity(intent)
                } else {
                    newvalue = 1
                    val intent = Intent(context, context.javaClass)
                    intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
                    intent.putExtra("value", newvalue.toString())
                    context.startActivity(intent)
                }
            }
        }.start()
        startBtn.text = "Pause"
        mTimeRunning = true
    }

    fun isRunning(): Boolean {
        return mTimeRunning
    }

    private fun updateTimer() {
        val minutes = (mTimeLeftInMillis / 1000) / 60
        val seconds = (mTimeLeftInMillis / 1000) % 60

        val timeLeftText = String.format("%02d:%02d", minutes, seconds)
        mtextView.text = timeLeftText
    }
}
